src/table/components/virtualized-table/Table.tsx

```diff
@@ -71,48 +75,46 @@ const TableContainer = (props: TableContainerProps) => {
   );
 
   return (
-    <div
-      data-testid="table-container"
+    <ScrollableContainer
       ref={ref}
-      style={{
-        overflow: constraints.active ? 'hidden' : 'auto',
-        width: rect.width,
-        height: rect.height - (paginationNeeded ? PAGINATION_HEIGHT : 0),
-      }}
+      constraints={constraints}
+      width={tableRect.width}
+      height={tableRect.height}
       onScroll={scrollHandler}
     >
       <FullSizeContainer width={containerWidth} height={containerHeight}>
-        <Header
-          headerStyle={headerStyle}
-          layout={layout}
-          rect={rect}
-          pageInfo={pageInfo}
-          columns={columns}
-          columnWidth={width}
-          forwardRef={headerRef}
-        />
-        {totals.atTop ? TotalsComponent : null}
-        <Body
-          bodyStyle={bodyStyle}
-          model={model}
-          layout={layout}
-          rect={rect}
-          pageInfo={pageInfo}
-          paginationNeeded={paginationNeeded}
-          columns={columns}
-          columnWidth={width}
-          forwardRef={bodyRef}
-          innerForwardRef={innerForwardRef}
-          selectionsAPI={selectionsAPI}
-          totals={totals}
-        />
-        {totals.atBottom ? TotalsComponent : null}
+        <StickyContainer rect={stickyContainerRect}>
+          <Header
+            headerStyle={headerStyle}
+            layout={layout}
+            rect={stickyContainerRect}
+            pageInfo={pageInfo}
+            columns={columns}
+            columnWidth={width}
+            forwardRef={headerRef}
+          />
+          {totals.atTop ? TotalsComponent : null}
+          <Body
+            bodyStyle={bodyStyle}
+            model={model}
+            layout={layout}
+            rect={stickyContainerRect}
+            pageInfo={pageInfo}
+            columns={columns}
+            columnWidth={width}
+            forwardRef={bodyRef}
+            innerForwardRef={innerForwardRef}
+            selectionsAPI={selectionsAPI}
+            totals={totals}
+          />
+          {totals.atBottom ? TotalsComponent : null}
+        </StickyContainer>
       </FullSizeContainer>
-    </div>
+    </ScrollableContainer>
   );
 };
 
 // Export non memoized version for testing purpose
-export { TableContainer };
+export { Table as TestableTable };
```

Comment:
why the casting?

Reply:
It is not casted but exported as a non memoized version for testing purpose